execution/test_utils.go

```diff
@@ -237,6 +276,55 @@ func ReadAll(ctx context.Context, stateStorage storage.Storage, stream RecordStr
 	return records, nil
 }
 
+func ReadAllWithCount(ctx context.Context, stateStorage storage.Storage, stream RecordStream, count int) ([]*Record, error) {
+	var records []*Record
+	for i := 0; i < count; {
+		tx := stateStorage.BeginTransaction()
+		ctx := storage.InjectStateTransaction(ctx, tx)
+
+		rec, err := stream.Next(ctx)
+		if err == ErrEndOfStream {
+			err := tx.Commit()
+			if err != nil {
+				return nil, errors.Wrap(err, "couldn't commit transaction")
+			}
+			break
+		} else if errors.Cause(err) == ErrNewTransactionRequired {
+			err := tx.Commit()
```

Comment:
I guess this case could be rewritten to
`_ := tx.Commit() 
continue`

Reply:
changed